### DISCUSSION TITLE
is there a embeddable debug macro for SLANG?

### DISCUSSION THREAD

**Discussion by Practical-UVM-Step-By-Step** [MikePopoloski/slang#871] [General]
You probably have it, but I'm not sure where to look. 

Would you by any chance have a macro which we can embed some strings and it spits out using the diag engine or other mechansim. Something we can use anywhere in the slang library as we're debugging because the namespaces etc are maintained?

if not, could you suggest such a macro?

Currently, i'm adding namespace std/#include <iostream> and then doing std::cout prints while I learn this library, and it's a bit tedious...

Srivatsa

**Comment by MikePopoloski** [comment]
There is not a debug macro that will print to the console. I tend to just use a debugger so I have no need of such a macro. If you prefer printf debugging then I'm not sure why it's more tedious to put the printfs in yourself than using a macro, but you're always free to add such a macro in your own fork.

**Comment by Practical-UVM-Step-By-Step** [comment]
Thanks Mike for the response. Let me see if I can do something with it.
